torch/csrc/jit/tensorexpr/llvm_codegen.cpp

```diff
@@ -826,7 +827,21 @@ void LLVMCodeGenImpl::visit(const Cast* v) {
       } else {
         value_ = irb_.CreateSIToFP(value_, dstType);
       }
+
     } else if (dstType->isIntOrIntVectorTy()) {
+      // Bool is represented as 8bits, we need to make sure
```

Comment:
I find this comment (and the description) a bit misleading. Any of the 8 bits is good-enough as far as true / false goes. We only hit problems when we observe the exact underlying value through a cast to integer.

I'd prefer a phrasing which focuses on the fact that we want true to be exactly 1 (as opposed to any non-zero value), that'd reflect better what's being fixed here.

Reply:
How is 
> we're appropriately setting the one bit that is used

different from 

> we want true to be exactly 1 

? I'm also not really sure that `Any of the 8 bits is good-enough as far as true/false goes` is true. Pytorch only uses the one bit. That is why the test was failing previously, because only the second bit was being set for `True + True`.

Reply:
When we add two `True` bools, we still get the correct *boolean* value, `True` on master. I printed the values in the test. It fails because the subsequent integer conversion yielded 2 instead of 1. Reading this comment, my guess would be that we lost all set bits and ended up with a `False` instead of a `True`.

Reply:
For me the original comment was also clear with regard that "the one bit that is used" meant that just promoting to ints and doing arithmetic there was not enough and led to incorrect results. It might be worth adding the 1+1 example just to make it crystal clear though.